### DISCUSSION TITLE
Anomaly prediction interval with pretrained models.

### DISCUSSION THREAD

**Discussion by nagydavid** [winedarksea/AutoTS#228] [Q&A]
HI All, 

Is there a possibility to use already pretrained models with the anomaly_detector[prediction_interval] method? if yes, I cannot find the way to supply the params to the detector.

since I have 300+ ts I would not do the training twice.

thanks in advance.

**Comment by winedarksea** [comment]
Firstly, the prediction interval approach is always going to be the slowest. If speed is really a concern the other approaches are going to work better (and in my experience usually catch anomalies better too).

Have you tried `AnomalyDetector.get_new_params()` it generates random parameters. It should give you the parameter input styles available. Just keep running it in a loop and plotting the results until you get something you like.

It does not currently exist but I could probably modify the function to allow a manual, externally generated backcast to be input instead of running the model. If that is what you really need.

**Reply by nagydavid** [reply]
thanks, right now, I am planning to looping it through with
```
   from autots.tools.anomaly_utils import limits_to_anomalies

    limits_to_anomalies(forecast["forecast"], forecast["upper_forecast"], forecast["lower_forecast"])
```
as I see thats the one which generates the anomaly scores, based on the forecast values.

the method parameter is only prediction_interval?

can it work?

**Reply by winedarksea** [reply]
The prediction interval approach does have some other optional parameters, you can see them in the function.
https://github.com/winedarksea/AutoTS/blob/3e6baffebed6fd017f496016248de3d2907456aa/autots/tools/anomaly_utils.py#L164

It is taking the zscore of the observed - upper or lower forecast and using zscore anomaly detection on that. So the quality of your upper and lower bound forecasts will have a big impact on the anomaly detection.


The other methods don't require a forecast but still generate anomaly scores. I'll say it again, they are probably going to be better in most cases.

**Reply by nagydavid** [reply]
thanks

**Comment by winedarksea** [comment]
> thanks, right now, I am planning to looping it through with
> 

>     limits_to_anomalies(forecast["forecast"], forecast["upper_forecast"], forecast["lower_forecast"])

I think you want forecast['forecast'] to be replaced with df (observed not forecast values) here
